base/bucket.go

```diff
@@ -32,32 +32,68 @@ const (
 )
 
 const (
-	GoCouchbase CouchbaseDriver = iota
-	GoCB
-	GoCBGoCouchbaseHybrid // Defaults to GoCB, falls back to GoCouchbase for missing functionality
+	GoCouchbase            CouchbaseDriver = iota
+	GoCB                                   // Use GoCB driver with default Transcoder
+	GoCBCustomSGTranscoder                 // Use GoCB driver with a custom Transcoder
 )
 
 const (
 	DataBucket CouchbaseBucketType = iota
 	IndexBucket
 )
 
-var (
-	DefaultDriverForBucketType = map[CouchbaseBucketType]CouchbaseDriver{
-		DataBucket: GoCBGoCouchbaseHybrid,
-		// DataBucket:  GoCouchbase,
-		IndexBucket: GoCB,
-	}
+const (
+	ViewQueryParamStale         = "stale"
+	ViewQueryParamReduce        = "reduce"
+	ViewQueryParamStartKey      = "startkey"
+	ViewQueryParamEndKey        = "endkey"
+	ViewQueryParamInclusiveEnd  = "inclusive_end"
+	ViewQueryParamLimit         = "limit"
+	ViewQueryParamIncludeDocs   = "include_docs"
+	ViewQueryParamDescending    = "descending"
+	ViewQueryParamGroup         = "group"
+	ViewQueryParamSkip          = "skip"
+	ViewQueryParamGroupLevel    = "group_level"
+	ViewQueryParamStartKeyDocId = "startkey_docid"
+	ViewQueryParamEndKeyDocId   = "endkey_docid"
+	ViewQueryParamKey           = "key"
+	ViewQueryParamKeys          = "keys"
 )
 
+func ChooseCouchbaseDriver(bucketType CouchbaseBucketType, feedType string) CouchbaseDriver {
+
+	feedType = strings.ToLower(feedType)
+	switch feedType {
+	case DcpShardFeedType:
+		// If this is a DCPSHARD feed, must use go-couchbase, since VerifyBucketSequenceParity() expects
+		// a base.CouchbaseBucket.  Ditto for code that interfaces w/ CBGT
```

Comment:
Ok so looks like this change was motivated by https://github.com/couchbase/sync_gateway/issues/2423#issuecomment-300002305:

> sg-accel was using GoCB (with CustomSGTranscoder) to open the data-bucket, even though the feedtype was set to DCPSHARD

and `VerifyBucketSequenceParity` was trying to type assert the bucket to a `CouchbaseBucket` for no good reason, and giving that error.

I removed VerifyBucketSequenceParityCouchbaseBucket() in  be24756acfd7f43acc1323468e25974918eb8c17, and will look into removing the `case DcpShardFeedType:` as well.